ci/build_common.sh

```diff
@@ -6,109 +6,171 @@ set -eo pipefail
 cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )";
 
 # Script defaults
+HOST_COMPILER=${CXX:-g++} # $CXX if set, otherwise `g++`
+CXX_STANDARD=17
 CUDA_COMPILER=nvcc
+CUDA_ARCHS= # Empty, use presets by default.
 
 # Check if the correct number of arguments has been provided
 function usage {
-    echo "Usage: $0 [OPTIONS] <HOST_COMPILER> <CXX_STANDARD> <GPU_ARCHS>"
+    echo "Usage: $0 [OPTIONS]"
+    echo
     echo "The PARALLEL_LEVEL environment variable controls the amount of build parallelism. Default is the number of cores."
-    echo "Example: PARALLEL_LEVEL=8 $0 g++-8 14 \"70\" "
-    echo "Example: $0 clang++-8 17 \"70;75;80-virtual\" "
-    echo "Possible options: "
-    echo "  -nvcc: path/to/nvcc"
+    echo
+    echo "Options:"
     echo "  -v/--verbose: enable shell echo for debugging"
+    echo "  -nvcc: path/to/nvcc"
```

Comment:
suggestion: Since we're redoing this all anyways, I think we should rename `-nvcc` to `--cuda` to better support using clang as the CUDA compiler. I did a short term workaround in https://github.com/NVIDIA/cccl/pull/493 but it'd be nice to do it right.

Reply:
Done.